plugins/modules/mso_schema_template_externalepg.py

```diff
@@ -228,13 +253,15 @@ def main():
     elif state == 'present':
         vrf_ref = mso.make_reference(vrf, 'vrf', schema_id, template)
         l3out_ref = mso.make_reference(l3out, 'l3out', schema_id, template)
+        anp_ref = mso.make_reference(anp, 'anp', schema_id, template)
         if display_name is None and not mso.existing:
             display_name = externalepg
 
         payload = dict(
             name=externalepg,
             displayName=display_name,
             vrfRef=vrf_ref,
+            anpRef=anp_ref,
```

Comment:
don't always add it to payload.
Remove it from here and add a line after payload like:
if anp is not None:
    payload['anpRef'] = anp_ref

Reply:
Done!